### ISSUE TITLE
[networkpolicyDB] "security_groups_id_seq" reached maximum value

### ISSUE THREAD

**Issue by petrinski** [cloudfoundry/cf-networking-release#223]
In one of our environments, we encountered an error related to the security_groups table in the network-policy DB.
The error message indicates that the nextval (next value) has reached the maximum value of the sequence called "security_groups_id_seq."

```
nasg-syncer exited with error: 
saving security group ... nextval: reached maximum value of sequence \"security_groups_id_seq\" (2147483647)\nasg-lock exited with nil\n"} 
```

The security_group id itself and the sequence are of type int with max value of 2147483647.
The asg-syncer executes a sync (not sure why the method is called [Replace](https://github.com/cloudfoundry/cf-networking-release/blob/develop/src/code.cloudfoundry.org/policy-server/store/security_groups_store.go#L122)) at regular intervals, which inserts new rules. As to the old ones, we check if a security_group with guid P exists, if it does, we execute a UPSERT operation. 
 
If you have a large number of security groups, it's possible to exceed the maximum value, since every time we execute the sync, we update all the entries. 

For instance, if we set a time period of 2 minutes and let's say we have 20,000 security groups, every two minutes, we will execute the sync. As a result, the sequences will increase by 20,000 every 2 minutes.
To mitigate this problem, we see two possible approaches.

First, you can replace the data type of the id and sequence  from int to bigint, which will give us a much bigger limit for the number of sequences.

And second one, you can modify the upsert logic in the code mentioned below, in a way that we do not increate the sequences every time the sync is executed.

**Comment by geofffranks** [comment]
Let's go with the bigint data type. Should allow for 1 million years of syncing every second with hundreds of thousands of ASGs. It doesn't require a lot of thought on potential downsides. Increased DB footprint would be 4 bytes per ASG which is pretty negligible for the volume of ASGs in a database.

**Comment by ZPascal** [comment]
Hi @geofffranks & @petrinski,

it would also follow the same schema as the other table sequences inside this database and round up the overall view for this database in general.
![image (1)](https://github.com/cloudfoundry/cf-networking-release/assets/13447634/1a9f8969-b9f3-4298-9e7f-c848248164a2)

I already opened last week a draft [PR](https://github.com/cloudfoundry/cf-networking-release/pull/222 ) to solve the case. I'm not sure if anything else is necessary to handle it. It would be nice if you could validate it.

**Comment by ZPascal** [comment]
Thanks, @geofffranks, for merging the PR. I think we can close this issue.

**Comment by geofffranks** [comment]
This has been released in v3.31.0 of cf-networking!
